openedx/features/redhouse_panel/api/v0/views.py

```diff
@@ -1,16 +1,96 @@
+from django.contrib.auth import get_user_model
 from django.contrib.sites.models import Site
+from django.db.models import F, Q, Count, Case, When, OuterRef, Exists, BooleanField
 from django.shortcuts import get_object_or_404
-from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
+from rest_framework import status
 from rest_framework.authentication import SessionAuthentication
 from rest_framework.permissions import IsAuthenticated
 from rest_framework.response import Response
 from rest_framework.views import APIView
 from rest_framework_oauth.authentication import OAuth2Authentication
 
+from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
 from openedx.core.djangoapps.site_configuration.models import SiteConfiguration
+from openedx.core.djangoapps.theming.helpers import get_current_site
 from openedx.core.lib.api.permissions import IsStaffOrOwner
-
 from openedx.features.redhouse_panel.api.v0.serializers import SiteSerializer
+from student.models import CourseAccessRole
+from student.roles import CourseStaffRole, CourseInstructorRole
+
+User = get_user_model()
+
+
+class UsersAccountStatsAPIView(APIView):
+    authentication_classes = (
+        JwtAuthentication,
+        OAuth2Authentication,
+        SessionAuthentication,
+    )
+    permission_classes = (IsAuthenticated, IsStaffOrOwner,)
+
+    default_stats = {
+        'instructor_count': 0,
+        'student_count': 0
+    }
+
+    def get(self, request):
+        site = get_current_site()
+
+        site_users_query = Q(is_staff=True) | \
+                           Q(edly_profile__edly_sub_organizations__lms_site=site) | \
+                           Q(edly_profile__edly_sub_organizations__studio_site=site) | \
+                           Q(edly_profile__edly_sub_organizations__preview_site=site)
+
+        is_instructor_query = Exists(
+            CourseAccessRole.objects.filter(
+                user=OuterRef('id'),
+                role__in=[CourseStaffRole.ROLE, CourseInstructorRole.ROLE]
+            )
+        )
+
+        instructor_count_query = Count(
+            Case(
+                When(is_instructor=True, then=1)
+            )
+        )
+
+        student_count_query = Count(
+            Case(
+                When(is_instructor=False, then=1)
+            )
+        )
+
+        _stats = get_user_model().objects.filter(
```

Comment:
We already have `User` variable at global level in this file. We can use that here instead of calling the `get_user_model()` again.

Please see line no 20.